src/Provider/DefaultSmsProvider.php

```diff
@@ -216,6 +203,8 @@ public function processDeliveryReport(Request $request, SmsGatewayInterface $sms
    * @return \Drupal\sms\Entity\SmsGatewayInterface|null
    *   A SmsGateway config entity, or NULL if default gateway is not set or
    *   invalid.
+   *
+   * @deprecated remove. shouldnt rely on this since its really a fallback.
```

Comment:
Is it really a fallback. This is a setting that comes from the UI and I think we need to retain it for users who don't need the complex setups.